cmd/incus-simplestreams/main_prune.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"slices"
+	"sort"
+
+	"github.com/spf13/cobra"
+
+	cli "github.com/lxc/incus/v6/internal/cmd"
+	"github.com/lxc/incus/v6/shared/simplestreams"
+)
+
+type cmdPrune struct {
+	global *cmdGlobal
+
+	flagDryRun        bool
+	flagPruneVersions bool
+	flagVerbose       bool
+}
+
+// Command generates the command definition.
+func (c *cmdPrune) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = "prune"
+	cmd.Short = "Clean up obsolete files and data"
+	cmd.Long = cli.FormatSection("Description",
+		`Cleans up obsolote tarball files and removes outdated versions of a product
+
+The prune command scans the project directory for tarball files that do not have corresponding references
+in the 'images.json' file. Any tarball file that is not listed in images.json is considered orphaned
+and will be deleted.
+The prune command identifies and deletes older versions of product data,
+retaining only the most recent existing version.`)
```

Comment:
That last paragraph and the fact that the `prune` command has `prune-versions` flag is pretty confusing.

I think I'd rephrase that last paragraph as:
```

Additionally this command will delete older images, keeping a configurable number of older images per product.
```

Then instead of `prune-version` we should probably just have a `--retention` flag which defaults to `2` as in we'll keep two older versions of the product by default (same as we do on the public image server).

Reply:
So running `incus-simplestreams prune` should:
 - Find any orphaned files (no record in the index) and delete them
 - Find any broken image and delete them from the index
 - Find any product that has more than 1+flagRetention images and delete the older ones

Reply:
What do you mean by `broken image`?

Reply:
That would be an image that's in the index but for which some of the files are missing on disk.

Reply:
Done.